data-prepper-api/src/test/java/org/opensearch/dataprepper/model/codec/JsonDecoderTest.java

```diff
@@ -60,15 +69,93 @@ void test_basicJsonDecoder_withTimeReceived() {
         try {
             jsonDecoder.parse(new ByteArrayInputStream(inputString.getBytes()), now, (record) -> {
                 receivedRecord = record;
-                receivedTime = ((DefaultEventHandle)(((Event)record.getData()).getEventHandle())).getInternalOriginationTime();
+                receivedTime = record.getData().getEventHandle().getInternalOriginationTime();
             });
         } catch (Exception e){}
-        
+
         assertNotEquals(receivedRecord, null);
         Map<String, Object> map = receivedRecord.getData().toMap();
         assertThat(map.get("key1"), equalTo(stringValue));
         assertThat(map.get("key2"), equalTo(intValue));
         assertThat(receivedTime, equalTo(now));
     }
 
+    @Nested
+    class JsonDecoderWithInputConfig {
+        private ObjectMapper objectMapper;
+        final List<String> include_keys = new ArrayList<>();
+
+        @BeforeEach
+        void setup() {
+            objectMapper = new ObjectMapper();
+        }
+        @Test
+        void test_basicJsonDecoder_withInputConfig() throws IOException {
+            Random r = new Random();
+            final Instant now = Instant.now();
+            List<Record<Event>> records = new ArrayList<>();
+            for (int i=0; i<10; i++) {
+                include_keys.add(UUID.randomUUID().toString());
+            }
+            final String key_name = "logEvents";
+            Map<String, Object> jsonObject = generateJsonWithSpecificKeys(include_keys, key_name, 10);
+            jsonDecoder = new JsonDecoder(key_name, include_keys, include_keys);
+            jsonDecoder.parse(createInputStream(jsonObject), now, (record) -> {
+                records.add(record);
+                receivedTime = ((Event)record.getData()).getEventHandle().getInternalOriginationTime();
+            });
+
+            records.forEach(record -> {
+                Map<String, Object> dataMap = record.getData().toMap();
+                Map<String, Object> metadataMap = record.getData().getMetadata().getAttributes();
+
+                for (String include_key: include_keys) {
```

Comment:
Assert the size of `include_keys` first or you may miss an empty list. Also, rename to `includeKeys`.